testsuite/tests/typing-missing-cmi-2/test.ml

```diff
@@ -1,15 +1,17 @@
 (* TEST
 files = "foo.mli bar.mli baz.ml"
-* setup-ocamlc.byte-build-env
-** ocamlc.byte
-module = "foo.mli"
+* skip
+reason = "Not supporting this use of missing cmis with layouts"
```

Comment:
Not really.  (As we've discussed, a consequence of the layout system is that we can support missing `cmi`s in fewer cases, because previously an abstract type (due to missing cmi) was just an abstract type, but now we might need to know its layout.  When the definition is missing, we have to guess `any` for soundness.)

I suspect the right place to document this kind of user-facing change in expectations about what needs to be passed with `-I` is in the design documentation and release notes.  There's not really a specific place in the compiler where we've changed this - it's a consequence of layout checks broadly.